lib/puppet/type/k5login.rb

```diff
@@ -78,6 +128,54 @@ def mode=(value)
       File.chmod(Integer("0#{value}"), @resource[:name])
     end
 
+    # Set the file seluser
+    def seluser
+      context = get_selinux_current_context(@resource[:name])
+      return context[:seluser]
```

Comment:
`context` here is a string, not a hash. You need to run this through `parse_selinux_context`:
```
return parse_selinux_context(:seluser, context)
```